google/cloud/spanner_v1/table.py

```diff
@@ -43,13 +43,26 @@ class Table(object):
     :param database: The database that owns the table.
     """
 
-    def __init__(self, table_id, database):
+    def __init__(self, table_id, database, schema_name=None):
```

Comment:
Yes:
1. In `list_tables` the current behavior is to list all tables in the default schema (so for GoogleSQL, in the schema named 'empty string'). We want to retain that behavior, so we don't make a breaking change for anyone.
2. `list_tables` should also be able to return all tables across all schemas. To do this, you can pass in `None`. That means that we cannot use `None` to specify the default behavior that we want.
3. The name of the default schema is different in PostgreSQL and GoogleSQL. So for `list_tables` we need an additional special value to indicate 'use the default schema of this database', which cannot be `None` (because point 2), and can also not be for example the empty string (because for PostgreSQL it would be something else).
4. Here, we don't need two different special values, so we can use `None` as the default, meaning that 'the user did not specify any schema'. That's then also what we should use in queries that use this table, and then Cloud Spanner will decide what it means when it receives an unqualified table name (a table name without a schema name).

Reply:
Got it